core/src/datayoga_core/blocks/parquet/write/block.py

```diff
@@ -0,0 +1,40 @@
+
+import logging
+import os
+from abc import ABCMeta
+from typing import Any, Dict, List, Optional
+
+from datayoga_core import utils
+from datayoga_core.block import Block as DyBlock
+from datayoga_core.context import Context
+from datayoga_core.result import BlockResult
+from fastparquet import write
+from pandas import DataFrame
+
+logger = logging.getLogger("dy")
+
+
+class Block(DyBlock, metaclass=ABCMeta):
+
+    def init(self, context: Optional[Context] = None):
+        logger.debug(f"Initializing {self.get_block_name()}")
+        parquet_file = self.properties["file"]
+
+        if os.path.isabs(parquet_file) or context is None:
+            self.file = parquet_file
+        else:
+            self.file = os.path.join(context.properties.get("data_path"), parquet_file)
+
+        logger.debug(f"file: {self.file}")
+
+    async def run(self, data: List[Dict[str, Any]]) -> BlockResult:
+        logger.debug("Writing parquet")
+
+        # remove the internal $$msg_id column
+        out = [dict(data={i: record[i] for i in record if i != Block.MSG_ID_FIELD}) for record in data]
```

Comment:
I think we should have a utility for this, as we use it in other places.

Reply:
yes, let's open another issue for this - but for now, as said, it would be better to have such a utility rather code duplication.

Reply:
For now:
In std/write/block.py:
```
# remove the internal $$msg_id column
filtered_record = {i: record[i] for i in record if i != Block.MSG_ID_FIELD}
```
we should have a utility for this with the code you changed in parquet/writ/block.py

Reply:
✅